### PR TITLE
Add support for custom paths for layout and template files

### PR DESCRIPTION
**Pull request by Spriz**

### CODE REVIEW

File: src/Pdf/CakePdf.php [hunk 1]
```diff
@@ -738,6 +752,40 @@ public function template($template = false, $layout = null)
     }
 
     /**
+     * Template path
+     *
+     * @param mixed $templatePath The path of the template to use
+     * @return mixed
+     */
+    public function templatePath($templatePath = false)
```
Comment (dereuromark) [inline src/Pdf/CakePdf.php]: I wonder if null would be a better default null value here. Then docblock: string|null

Comment (Spriz) [inline src/Pdf/CakePdf.php]: Might be, i just used took same approach as rest, where if use null to delete the current value, false to get the current value, and string to set new value - so I did this for consistency.

Comment (Spriz) [inline src/Pdf/CakePdf.php]: Comments @dereuromark? Should I change it to null as default? :)

Comment (dereuromark) [inline src/Pdf/CakePdf.php]: If that works, that sounds like a cleaner API to me.

Comment (Spriz) [inline src/Pdf/CakePdf.php]: should I change the others too then? (https://github.com/FriendsOfCake/CakePdf/blob/master/src/Pdf/CakePdf.php#L726)

Comment (dereuromark) [inline src/Pdf/CakePdf.php]: Autsch, thats not good. We cannot change it due to BC, so your change is OK for now.
Cleanup has to be done in a future release.

Comment (Spriz) [inline src/Pdf/CakePdf.php]: Yeah sorry, that's what I meant, change this PR to be like others for consistency, and make a PR afterwards that cleans it up. The one can me merged now, and the other can go pending for next major release.

Comment (dereuromark) [inline src/Pdf/CakePdf.php]: :+1:

**Comment by dereuromark** [comment]
Good to go!

**Comment by Spriz** [comment]
Sweet @dereuromark - just poke me if anything's needed.

**Comment by dereuromark** [comment]
Why does it not show the travis build?

**Comment by Spriz** [comment]
No clue, maybe Travis is only enabled for collaborators?

**Comment by Spriz** [comment]
@dereuromark anything I could do?

**Comment by dereuromark** [comment]
You can check if any of the tests fail after the merge and help to fix them :) That would be awesome.

**Comment by Spriz** [comment]
Seems all good: https://travis-ci.org/FriendsOfCake/CakePdf/builds/106972697

**Pull request merged**